src/plugins/expressions/common/expression_functions/types.ts

```diff
@@ -87,6 +94,33 @@ export interface ExpressionFunctionDefinition<
    */
   fn(input: Input, args: Arguments, context: Context): Output;
 
+  /**
+   * migrate function
+   * @param state any previuous or current version of the ast for this function
+   * @param version string representation of version of the state
+   */
+  migrate?(state: unknown, version: string): any;
```

Comment:
Is there any reason we would want/need `migrate` to be async?

I know this need has come up for SO migrations, but we made the explicit decision not to enable this so that one bad migration wouldn't slow everything down, but just wanted to ask.

FWIW I think keeping it sync is most likely fine, just wanted to raise the question.

Reply:
i would prefer to keep it sync unless we have an actual usecase for async